src/Cardano/Shell/NodeIPC/ServerExample.hs

```diff
@@ -101,13 +100,20 @@ exampleServerWithProcess msg = bracket acquire restore (action msg)
         hClose wHandle
         unsetEnv "NODE_CHANNEL_FD"
 
-    action :: MsgIn -> (ReadHandle, Handle) -> IO (MsgOut, MsgOut)
+    action :: MsgIn
+           -> (ReadHandle, Handle)
+           -> IO (Either NodeIPCException (MsgOut, MsgOut))
     action msgin (readHandle, _) = do
-        withCreateProcess (proc "stack" ["exec", "node-ipc", "haskell"])
-            { std_in = CreatePipe } $
-                \(Just stdIn) _ _ _ -> do
+        -- For some reason, @--no-nix@ argument is required for this function to run
+        -- currently asking DevOps regarding this.
+        withCreateProcess (proc "stack" ["exec", "--no-nix", "node-ipc", "haskell"])
```

Comment:
I don't think its a good idea to make this *require* `stack` and/or `nix.

Reply:
to run the example from the source directory this looks like a reasonable way to do it.

Reply:
also, using `stack exec node-ipc` breaks the ability to pass file-handles to the child process, which is why the `js` vs `haskell` flag exists here

`$(stack path --local-install-root)/bin/node-ipc` will return the path to the `node-ipc` binary made by stack, allowing you to run it in `js` mode

so the only missing part, is to either ensure the `ServerExample.hs` is always ran with `$(stack path --local-install-root)/bin` added to `PATH` (then you can just assume `node-ipc` is a valid cmd)
or you need to try `node-ipc` and then fallback to asking stack where it was